kernel/kernel-api/src/main/java/io/delta/kernel/internal/metrics/hook/PostCommitHookExecutionContext.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.metrics.hook;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.hook.PostCommitHook.PostCommitHookType;
+import io.delta.kernel.metrics.PostCommitHookMetricsResult;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+/** Encapsulates context for a post-commit hook execution. */
+public class PostCommitHookExecutionContext {
+
+  private final String tablePath;
+  private final PostCommitHookType hookType;
+  private final Map<String, String> hookExecutionDetails;
```

Comment:
Alternate strawman, this is just an `Object`. Hooks can put whatever they want there and then cast it back.